src/compute/mean.ts

```diff
@@ -1,19 +1,37 @@
 import { Image } from '../Image';
+import { Point } from '../geometry';
+
+export interface MeanOptions {
+  /**
+   * Points to calculate mean from.
+   */
+  points: Point[];
+}
 
 /**
  * Compute the mean of an image. The mean can be either computed on each channel
  * individually or on the whole image.
  * @param image - Image to process.
+ * @param options - Mean options.
  * @returns The mean pixel.
  */
-export function mean(image: Image): number[] {
+export function mean(image: Image, options?: MeanOptions): number[] {
   const pixel = new Array<number>(image.channels).fill(0);
-  for (let row = 0; row < image.height; row++) {
-    for (let column = 0; column < image.width; column++) {
+  const nbValues = options ? options.points.length : image.size;
+  if (options) {
+    for (const point of options.points) {
       for (let channel = 0; channel < image.channels; channel++) {
-        pixel[channel] += image.getValue(column, row, channel);
+        pixel[channel] += image.getValueByPoint(point, channel);
+      }
+    }
+  } else {
+    for (let row = 0; row < image.height; row++) {
+      for (let column = 0; column < image.width; column++) {
+        for (let channel = 0; channel < image.channels; channel++) {
+          pixel[channel] += image.getValue(column, row, channel);
+        }
       }
     }
   }
-  return pixel.map((channel) => channel / image.size);
+  return pixel.map((channel) => channel / nbValues);
```

Comment:
Why are you calling your sum "channel"

Reply:
Isn't it what it is? We find a mean value of a pixel. Each value in this pixel is a channel. Although, calling it `channelSum` can be a better option.